### PR TITLE
Allow sync tables to reuse schemas

### PR DESCRIPTION
**Pull request by patrick-codaio**
https://golinks.io/bug/23002

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,10 @@
+
+            Warning: The lock flag -r/--requirements will be deprecated in a future version
```
Comment (dweitzman-codaio) [inline requirements.txt:2]: Curious how if this warning was added manually or by some kind of tooling. Does it not need to be inside a `#`-style comment?

File: test/upload_validation_test.ts [hunk 2]
```diff
@@ -771,6 +771,60 @@ describe('Pack metadata Validation', () => {
...
+        const metadata = createFakePack({
+          syncTables: [table1, table2],
+        });
+        const validatedMetadata = await validateJson(metadata);
+
+        const {schema: schema1} = validatedMetadata.syncTables[0];
+        const {schema: schema2} = validatedMetadata.syncTables[1];
+        assert.ok(schema1);
+        assert.ok(schema2);
+        // assert.deepEqual(schema.identity, {name: 'SomeIdentity'});
```
Comment (dweitzman-codaio) [inline test/upload_validation_test.ts:825]: There's a commented-out line here. Did you intend to have this assert on the identity of the return values?

File: requirements.txt (hunk 1 above)
Comment (ekoleda-codaio) [reply requirements.txt:2]: This file is generated here:

https://github.com/coda/packs-sdk/blob/c3a47e4bfc19ac0fe9bb2da11f02fd31fcb6d7c6/Makefile#L42

Readthedocs requires it (it doesn't support Pipfile). Looks like this does [break the Readthedocs build](https://readthedocs.com/projects/coda-packs-sdk/builds/934991/).

**Review by jonathan-codaio — APPROVED** [review]

File: requirements.txt (hunk 1 above)
Comment (patrick-codaio) [reply requirements.txt]: Oops, sorry about that, removed for now. I'll take a stab in a different PR.

File: test/upload_validation_test.ts (hunk 2 above)
Comment (patrick-codaio) [reply test/upload_validation_test.ts]: Thank you for reminding me that I never finished the test!!

File: requirements.txt (hunk 1 above)
Comment (patrick-codaio) [reply requirements.txt]: https://github.com/coda/packs-sdk/pull/1972

**Pull request merged**